mmdet/core/nncf/compression.py

```diff
@@ -0,0 +1,122 @@
+    elif not cfg.nncf_load_from:
+        raise RuntimeError("Tried to load NNCF checkpoint, but there is no path")
+
+    if cfg.nncf_load_from:
+        resuming_state_dict = load_checkpoint(model, cfg.nncf_load_from)
+        logger.info(f"Loaded NNCF checkpoint from {cfg.nncf_load_from}")
+    else:
+        resuming_state_dict = None
+
+    def _get_fake_data_for_forward(cfg, nncf_config, get_fake_input_func):
+        # based on the method `export` of BaseDetector from mmdet/models/detectors/base.py
+        # and on the script tools/export.py
+        assert get_fake_input_func is not None
+
+        input_size = nncf_config.get("input_info").get('sample_size')
+        assert len(input_size) == 4 and input_size[0] == 1
+
+        H, W = input_size[-2:]
+        C = input_size[1]
+        orig_img_shape = tuple([H, W, C])  # HWC order here for np.zeros to emulate cv2.imread
+
+        device = next(model.parameters()).device
+
+        # NB: the full cfg is required here!
+        fake_data = get_fake_input_func(cfg, orig_img_shape=orig_img_shape, device=device)
+        return fake_data
+
+    def dummy_forward_without_export_part(model):
+        # based on the method `export` of BaseDetector from mmdet/models/detectors/base.py
+        # and on the script tools/export.py
+        fake_data = _get_fake_data_for_forward(cfg, nncf_config, get_fake_input_func)
+        img, img_metas = fake_data["img"], fake_data["img_metas"]
+        with model.forward_dummy_context(img_metas):
+            model(img)
+
+    def dummy_forward_with_export_part(model):
+        # based on the method `export` of BaseDetector from mmdet/models/detectors/base.py
+        # and on the script tools/export.py
+        fake_data = _get_fake_data_for_forward(cfg, nncf_config, get_fake_input_func)
+        img, img_metas = fake_data["img"], fake_data["img_metas"]
+        with model.forward_export_context(img_metas):
+            model(img)
+
+    if "nncf_should_use_dummy_forward_with_export_part" in cfg:
+        # TODO: this parameter is for debugging, remove it later
+        should_use_dummy_forward_with_export_part = cfg.get("nncf_should_use_dummy_forward_with_export_part")
+        logger.debug(f"set should_use_dummy_forward_with_export_part={should_use_dummy_forward_with_export_part}")
+
+    if should_use_dummy_forward_with_export_part:
+        logger.debug(f"dummy_forward = dummy_forward_with_export_part")
+        dummy_forward = dummy_forward_with_export_part
+    else:
+        logger.debug(f"dummy_forward = dummy_forward_without_export_part")
+        dummy_forward = dummy_forward_without_export_part
+
+    model.dummy_forward_fn = dummy_forward
+
+    compression_ctrl, model = create_compressed_model(model, nncf_config, dummy_forward_fn=dummy_forward,
+                                                      resuming_state_dict=resuming_state_dict)
+    return compression_ctrl, model
+
+
+def unwrap_nncf_model(module):
```

Comment:
This is a bit confusing that `wrap_nncf_model` changes the model object, while `unwrap_nncf_model` does not. Not sure that this is something bad, though.

Reply:
Agreed. We will consider it with @aleksei-kashapov and will try to find a better name.

Reply:
Good comment. I suggest to change the name from **unwrap_nncf_model** to **get_uncompressed_model**

Reply:
Done